.github/workflows/monthly-release.yaml

```diff
@@ -0,0 +1,90 @@
+on:
+  # Manual trigger mainly to create patch version
+  workflow_dispatch:
+    inputs:
+      patch_version:
+        description: 'to control the patch version of TinyTeX (i.e., z in x.y.z), so we could release patch versions like 2020.09.19 when necessary' 
+        required: false
+        type: string
+  # Monthly release 
+    schedule: 
+      - cron: '23 7 1 * *'
+
+    
+name: Publish Monthly Release
+
+jobs:
+  build-linux:
+    runs-on: ubuntu-latest
+    name: Publish Montly release
+
+    env:
+      TINYTEX_INSTALLER: TinyTeX
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      GH_REPO: ${{ github.repository }}
+
+    steps:
+      - name: Clone repo
+        uses: actions/checkout@v3
+        with:
+          # needed to get the tags
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config user.name 'github-actions[bot]'
+          git config user.email 'github-actions[bot]@users.noreply.github.com'
+
+      - name: Current tag to target
+        run: |
+          # Are we making a patch version ? 
+          patch_version=${{ github.event.inputs.patch_version }}
```

Comment:
Previously on AppVeyor, I was using an env var `TINYTEX_PATCH_VERSION` to set the patch version number. When I need to make a patch release, I manually set this env var and trigger the build (and unset the env var after the build is done). I wonder how I should make the patch release now?

Reply:
Yes the idea would be through the `workflow_dispatch` event configure here: 
https://github.com/yihui/tinytex-releases/blob/e08ca8ba1ff0e68331af804ae35ed61be2bac46a/.github/workflows/monthly-release.yaml#L3-L8
There is a `patch_version` input that can be set. 

Two ways to trigger this: 

* Either using `gh` CLI, [`gh workflow run`](https://cli.github.com/manual/gh_workflow_run) passing key value pair with a value of the day. Ex with today `-f patch_version=25` which will creaye `2022.04.25`
````sh
gh workflow run monthly-release.yaml -f patch_version=25
````

* Either using the menu in the Action pane - example in my fork 
  ![image](https://user-images.githubusercontent.com/6791940/165107848-07737f17-a417-41ed-aaaa-724bc6bce4c8.png)
  By default nothing is set = no patch version. If filled, then a patched version will be made. 

More docs: 
* https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/ 
* https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow

Reply:
Oh great! Thanks for teaching me this!